src/libPersistence/ContractStorage2.cpp

```diff
@@ -157,8 +157,18 @@ bool ContractStorage2::FetchStateValue(const dev::h160& addr, const bytes& src,
     return false;
   }
 
-  const auto& d_found = m_indexToBeDeleted.find(key);
-  if (d_found != m_indexToBeDeleted.end()) {
+  auto d_found = t_indexToBeDeleted.find(key);
+  if (d_found != t_indexToBeDeleted.end()) {
+    // ignore the deleted empty placeholder
+    if ((unsigned int)query.indices().size() == query.mapdepth()) {
+      foundVal = false;
+      return true;
+    }
+  }
+
+  d_found = m_indexToBeDeleted.find(key);
```

Comment:
no need of this statement?

Reply:
Needed I think, since `d_found` is set to something else in line 160

Reply:
Correct, ignore

Reply:
no need, they share same data structure